admin/lib/validator/check-window-v2/check-window-admin-end-date-validator.js

```diff
@@ -0,0 +1,51 @@
+'use strict'
+const moment = require('moment')
+
+const checkWindowErrorMessages = require('../../errors/check-window-v2')
+const dateService = require('../../../services/date.service')
+const XRegExp = require('xregexp')
+const { isEmpty, isInt } = require('validator')
+
+/**
+ * Validates check window admin end date data
+ * @param {Object} validationError
+ * @param {Object} adminEndDateData
+ */
+module.exports.validate = (validationError, adminEndDateData) => {
+  const currentDate = moment.utc()
+  const currentYear = currentDate.format('YYYY')
+  const adminEndDate = dateService.createUTCFromDayMonthYear(adminEndDateData.adminEndDay,
+    adminEndDateData.adminEndMonth, adminEndDateData.adminEndYear)
+  const maxDaysInMonth = adminEndDate ? moment(adminEndDate).daysInMonth() : 31
+
+  // Admin end day
+  const isAdminEndDayEmpty = isEmpty(adminEndDateData.adminEndDay.trim())
+  if (isAdminEndDayEmpty || (!isAdminEndDayEmpty && !isInt(adminEndDateData.adminEndDay, { min: 1, max: maxDaysInMonth }))) {
+    validationError.addError('adminEndDay', checkWindowErrorMessages.adminEndDayWrongDay)
+  }
+  if (!isAdminEndDayEmpty && !XRegExp('^[1-9]\\d{0,1}$').test(parseInt(adminEndDateData.adminEndDay))) {
+    validationError.addError('adminEndDay', checkWindowErrorMessages.adminEndDayInvalidChars)
+  }
+  // Admin end month
+  const isAdminEndMonthEmpty = isEmpty(adminEndDateData.adminEndMonth.trim())
+  if (isAdminEndMonthEmpty || (!isAdminEndMonthEmpty && !isInt(adminEndDateData.adminEndMonth, { min: 1, max: 12 }))) {
+    validationError.addError('adminEndMonth', checkWindowErrorMessages.adminEndMonthWrongDay)
+  }
+  if (!isAdminEndMonthEmpty && !XRegExp('^[1-9]\\d{0,1}$').test(parseInt(adminEndDateData.adminEndMonth))) {
+    validationError.addError('adminEndMonth', checkWindowErrorMessages.adminEndMonthInvalidChars)
+  }
+  // Admin end year
+  const isAdminEndYearEmpty = isEmpty(adminEndDateData.adminEndYear.trim())
+  if (isAdminEndYearEmpty || (!isAdminEndYearEmpty && !isInt(adminEndDateData.adminEndYear, { min: currentYear, max: (currentYear * 1 + 10) }))) {
+    validationError.addError('adminEndYear', checkWindowErrorMessages.adminEndYearWrongDay)
+  }
+  if (!isAdminEndYearEmpty && (!XRegExp('^[0-9]+$').test(adminEndDateData.adminEndYear) || adminEndDateData.adminEndYear.length !== 4)) {
+    validationError.addError('adminEndYear', checkWindowErrorMessages.adminEndYearInvalidChars)
+  }
+
+  // Admin end date
+  if (adminEndDate && moment(currentDate).isAfter(adminEndDate)) {
```

Comment:
Those should already be moment objects, doing `moment()` over them seems redundant (that happens in the other validators too) although this is not a ‘breaking’ thing?

Reply:
@danwilliams updated